### PR TITLE
FromStr for SegmentType

### PR DESCRIPTION
**Pull request by kockan**
Might have understood this wrong, but a first attempt at #3

### CODE REVIEW

File: src/segment_type.rs [hunk 1]
```diff
@@ -69,9 +69,27 @@ impl TryFrom<u8> for SegmentType {
...
+    /// Returns the segment type given the string representation.
+    ///
+    /// # Errors
+    ///
+    /// - If `SegmentType` not valid
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        if value.len() == 1 {
+            Self::try_from(value.chars().nth(0).unwrap())
+        } else {
+            Err(ReadStructureError::ReadSegmentMultipleSegments(value.to_owned()))
```
Comment (nh13) [inline src/segment_type.rs:84]: I think we use `ReadSegmentTypeInvalid` or we need a new error type for when the segment type is mis-specified (empty string or multiple characters).

**Review by nh13 — CHANGES_REQUESTED** [review]
This looks great, thank-you!  Can you fix up the one issue remaining?

File: src/segment_type.rs (hunk 1 above)
Comment (kockan) [reply src/segment_type.rs]: `ReadSegmentTypeInvalid` takes a `char`, so added a new (but very similar) error type `ReadSegmentTypeStringInvalid` for when the segment type if empty or multiple characters along with the relevant tests.

**Review by nh13 — APPROVED** [review]
This is great, thank-you.  I'll merge once checks pass.

**Comment by nh13** [comment]
Thank-you @kockan

**Pull request merged**